### ISSUE TITLE
mat-menu missing API to set aria-label

### ISSUE THREAD

**Issue by jelbourn** [angular/components#17952]
`<mat-menu>` doesn't have an API that lets the user set an `aria-label` on the `role="menu"` element. 

I'd like to fix this long term by moving to a different menu API, something like
```html
<mat-menu *matMenuDef>
  ...
</mat-menu>
```
(where `mat-menu` is the `role="menu"` element)

In the interim, we should just add an input for this.

**Comment by drstonebraker** [comment]
Note there are similar related attributes which should have a similar api, such as aria-labelledby and aria-describedby

**Comment by jelbourn** [comment]
Right- that's what I meant by the long-term fix; I don't want to add an explicit API for every attribute, but we do need a shorter-term workaround for just `aria-label`
